Prose/ProseLib/Sources/ConversationFeature/Chat/Chat.swift

```diff
@@ -153,31 +98,47 @@ struct ChatView: NSViewRepresentable {
     let contentController = WKUserContentController()
 
     // Set logged in user JID
-    contentController.addUserScript(self.setAccountJIDScript)
+    MessagingContext { jsScript, completion in
+      contentController.addUserScript(
+        WKUserScript(source: jsScript, injectionTime: .atDocumentEnd, forMainFrameOnly: true)
+      )
+      completion(nil, nil)
+    }.setAccountJID(self.viewStore.loggedInUserJID)
 
     let actions = ViewStore(self.store.stateless)
     // Allow right clicking messages
-    contentController.addEventHandler(
-      JSEventHandler(event: "message:actions:view", action: MessageAction.showMenu),
-      viewStore: actions
-    )
+    contentController.addMessageEventHandler(for: .showMenu) { result in
+      actions.send(.messageEvent(MessageEvent.showMenu, from: result))
+    }
     // Allow toggling reactions
-    contentController.addEventHandler(
-      JSEventHandler(event: "message:reactions:react", action: MessageAction.toggleReaction),
-      viewStore: actions
-    )
+    contentController.addMessageEventHandler(for: .toggleReaction) { result in
+      actions.send(.messageEvent(MessageEvent.toggleReaction, from: result))
+    }
     // Enable reactions picker shortcut
-    contentController.addEventHandler(
-      JSEventHandler(event: "message:reactions:view", action: MessageAction.showReactions),
-      viewStore: actions
-    )
+    contentController.addMessageEventHandler(for: .showReactions) { result in
+      actions.send(.messageEvent(MessageEvent.showReactions, from: result))
+    }
 
     let configuration = WKWebViewConfiguration()
     configuration.userContentController = contentController
 
     let webView = WKWebView(frame: .zero, configuration: configuration)
     webView.loadFileURL(Files.messagingHtml.url, allowingReadAccessTo: Files.messagingHtml.url)
 
+    context.coordinator.ffi = FFI { [weak webView] jsString, completion in
+      webView?.evaluateJavaScript(jsString) { res, error in
+        if let res = res {
+          logger.debug("JavaScript response: \(String(reflecting: res))")
+        }
```

Comment:
Why do we need a `printResult` flag when we're using `os_log`? Wouldn't it be better to [handle that in Console.app](https://support.apple.com/en-gb/guide/console/cnslbf30b61a/mac)?

Reply:
> Why do we need a `printResult` flag when we're using `os_log`?

The result (or even worse, the script) can be quite verbose, and we mostly never need it. It's just a simple way to add a useful log, like `.debugActions()` does on a Reducer for example.

Reply:
I believe that long term we benefit from using our tools properly. In this case os_log offers us different logging levels and categories by which we can filter without twiddling boolean knobs in our code. Best of all these filters can be saved on an individual level.

However I do understand that `print` statements can be helpful in quick (or intense) debugging sessions. I've disabled os_log output in the Xcode console so that we can see temporarily inserted `prints` without any noise. You can toggle that at anytime by going into the scheme settings and disabling the `OS_ACTIVITY_DT_MODE` environment variable.

Going forward I'm planning to also channel the core lib logs into the os_logs so that we have the one unified logging solution that we were promised. Unfortunately this is a bit more complicated and requires some restructuring of the Rust project since uniffi logs are hard-coded and I wasn't ready to shave that yak yet.

Reply:
Gosh darnit. Solved. Removed public visibility. os_log messages below `default` level are now hidden from the Xcode console instead of all. Which doesn't obscure messages in the Console.app during debugging.

Reply:
> I believe that long term we benefit from using our tools properly. In this case os_log offers us different logging levels and categories by which we can filter without twiddling boolean knobs in our code. Best of all these filters can be saved on an individual level.

I totally agree we should use proper logging tools the right way! IIRC OSLog can't do tracing yet (I'm not sure about this now), but we could really benefit from it and remove the ugly `[\(domain)]` I had inserted. I don't like it either, it's just more useful that not having it *for now*.

And about the boolean, it's just because I don't want to flood the user's logs with useless long logs. However, as I said [below](https://github.com/prose-im/prose-app-macos/pull/105#discussion_r943295014), `trace` is equivalent to `debug` and IIRC `debug` is not in release builds. So I might have made a mistake inserting the boolean. I just can't find the doc about log retention…

> However I do understand that print statements can be helpful in quick (or intense) debugging sessions.

Exactly, but I try never to commit them (as we shouldn't)

Reply:
Seems like [you answered to my question already](https://github.com/prose-im/prose-app-macos/pull/105#discussion_r943276307), debug logs are not persisted.